### PR TITLE
enable hot reload for demo in dev

### PR DESCRIPTION
**Pull request by ThieryMichel**

### CODE REVIEW

**Comment by fzaninotto** [comment]
can't the `npm link` technique from #4359 work here?

**Comment by ThieryMichel** [comment]
> can't the `npm link` technique from #4359 work here?


Actually, yes a `yarn link` does the job nicely. Guess I will just update the README.

**Comment by ThieryMichel** [comment]
> can't the `npm link` technique from #4359 work here?

Sadly no, the demo use the builded module, and so even linked we still need to build the module. Which is already the case without link.

**Comment by ThieryMichel** [comment]
The correct solution was to build the packages in watch mode while launching the demo, lerna does the rest.

File: packages/ra-core/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
         "build": "yarn run build-cjs && yarn run build-esm",
         "build-cjs": "rimraf ./lib && tsc",
         "build-esm": "rimraf ./esm && tsc --outDir esm --module es2015",
-        "watch": "rimraf ./lib && tsc --watch"
```
Comment (ThieryMichel) [inline packages/ra-core/package.json]: the demo need the esm package.

**Pull request merged**